lib/beautiful/colorful.lua

```diff
@@ -0,0 +1,1395 @@
+-- Reduces Saturation to 0
+-- @return A new color object representing the grayscaled color, or nil if invalid
+-- @usage local grayscaled_color = c:grayscale()
+function color:grayscale()
+    return color.from.hsla(self.hue, 0, self.lightness, self.alpha)
+end
+
+-- }}} HSL-adjusting functions
+
+-- {{{ Color harmony functions
+
+--- Returns the complement of a color
+-- Adds 180 to Hue (hue will self-adjust to be within [0,360])
+-- @return A new color object representing the complementary color, or nil if invalid
+-- @usage local complemented_color = c:complement()
+function color:complement()
+    local hue = self.hue + 180
+    return color.from.hsla(hue, self.saturation, self.lightness, self.alpha)
+end
+
+--- Returns the left and right analogous colors (30deg hue shift)
+-- @return A new color object representing a -30 degree hue shift of the color, or nil if invalid
+-- @return A new color object representing a +30 degree hue shift of the color, or nil if invalid
+-- @usage local left, right = c:analogous()
+function color:analogous()
+    local left = self.hue - 30
+    local right = self.hue + 30
+    return color.from.hsla(
+            left,
+            self.saturation,
+            self.lightness,
+            self.alpha),
+        color.from.hsla(
+            right,
+            self.saturation,
+            self.lightness,
+            self.alpha)
+end
+
+--- Returns the left and right complementary colors (30deg shift from
+--complementary color)
+-- @return A new color object representing the +210 degree hue shift (180 + 30)
+-- of the color, or nil if invalid
+-- @return A new color object representing the +150 degree hue shift (180 - 30)
+-- of the color, or nil if invalid
+-- @usage local left, right = c:split_complementary()
+function color:split_complementary()
+    local left = self.hue + 180 + 30
+    local right = self.hue + 180 - 30
+    return color.from.hsla(
+            left,
+            self.saturation,
+            self.lightness,
+            self.alpha),
+        color.from.hsla(
+            right,
+            self.saturation,
+            self.lightness,
+            self.alpha)
+end
+
+--- Returns the left and right triadic colors (120deg shift from base color)
+-- @return A new color object representing a +120 degree hue shift of the color, or nil if invalid
+-- @return A new color object representing a -120 degree hue shift of the color, or nil if invalid
+-- @usage local left, right = c:triadic()
+function color:triadic()
+    local left = self.hue + 120
+    local right = self.hue - 120
+    return color.from.hsla(
+            left,
+            self.saturation,
+            self.lightness,
+            self.alpha),
+        color.from.hsla(
+            right,
+            self.saturation,
+            self.lightness,
+            self.alpha)
+end
+
+--- Returns three colors making up the tetradic colors (base complement, base
+--+ 60deg, base + 60deg complement)
```

Comment:
So, complement is a 120 deg rotation... So we have here `+180`, `+60`, and `+60+180=+240`... Right?
Is this worth mentioning explicitly in the docs?

Reply:
Yeah that could be clarified.